enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -450,16 +436,27 @@ func (s *SociArtifactStore) indexImage(ctx context.Context, image v1.Image, conf
 	if err != nil {
 		return nil, nil, err
 	}
-	// Write the pointer from image identifier to SOCI Index in the ActionCache
-	// because the entry isn't content-addressable.
 	serializedIndexResourceName, err := indexResourceName.DownloadString()
 	if err != nil {
 		return nil, nil, err
 	}
-	err = s.env.GetCache().Set(ctx,
-		digest.NewResourceName(sociIndexCacheKey, "", rspb.CacheType_AC, repb.DigestFunction_SHA256).ToProto(),
-		[]byte(serializedIndexResourceName))
-	if err != nil {
+
+	// Store the SOCI Index as an ActionResult with:
+	// - The digest of the SOCI Index serialized in the stdout field
+	// - The digest of all artifacts (ZToCs and SOCI Index) as OutputFiles
+	var artifacts []*repb.OutputFile
+	artifacts = append(artifacts, &repb.OutputFile{Digest: indexResourceName.GetDigest()})
+	for _, ztocDigest := range ztocDigests {
+		artifacts = append(artifacts, &repb.OutputFile{Digest: ztocDigest})
+	}
+	req := repb.UpdateActionResultRequest{
+		ActionDigest: sociIndexCacheKey,
+		ActionResult: &repb.ActionResult{
+			StdoutRaw:   []byte(serializedIndexResourceName),
```

Comment:
I don't like overloading Stdout like this. What if we stored it as another output file, using path or node_properties to identify the type of data?

Reply:
SG, done